agent_integration_test.go

```diff
@@ -0,0 +1,64 @@
+//go:build integration
+// +build integration
+
+package tfe
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAgentsRead(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	agent, _, agentCleanup := createAgent(t, client, nil, nil, nil)
+	defer agentCleanup()
```

Comment:
We've begun using `t.Cleanup(agentCleanup)` in favor of `defer`. `t.Cleanup()` is useful for cleaning up resources that are allocated by helper functions since it ensures all subsequent subtests have finished running before invoking.

Reply:
@sebasslash Got it. Thanks for sharing that knowledge.